[Manufacturing] [Loitering Detection] [Smart Parking] [main] Use Intel Device Plugins for GPU/NPU access (#3010)
open-edge-platform/edge-ai-suites: manufacturing-ai-suite/industrial-edge-insights-vision/docs/user-guide/pallet-defect-detection/get-started/deploy-with-helm.md

## Prerequisites

- [System Requirements](../get-started/system-requirements.md)

> - K8s installation on single or multi node must be done as pre-requisite to continue the following deployment. Note: The kubernetes cluster is set up with `kubeadm`, `kubectl` and `kubelet` packages on single and multi nodes with `v1.30.2`.

Reviewer: ```suggestion
- K8s installation on single or multi node must be done as prerequisite to continue the following deployment. Note: The Kubernetes cluster is set up with `kubeadm`, `kubectl` and `kubelet` packages on single and multi nodes with `v1.30.2`.
```

Move what's in L59 here, so we don't have duplicate content.

>   Refer to tutorials online to setup kubernetes cluster on the web with host OS as ubuntu 22.04 and/or ubuntu 24.04.

Reviewer: ```suggestion
  Refer to tutorials online to setup Kubernetes cluster on the web with host OS as Ubuntu 22.04 and/or Ubuntu 24.04.
```
Move what's in L60 here, so we don't have duplicate content.

> - For helm installation, refer to [helm website](https://helm.sh/docs/intro/install/)

Reviewer: ```suggestion
- For Helm installation, refer to [Helm website](https://helm.sh/docs/intro/install/)
```
Moved from L61 to avoid duplication.

  kubectl apply -k "https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/nfd/overlays/node-feature-rules?ref=${RELEASE_VERSION}"

  # Step 6: Verify NFD pods are running
  kubectl get pods -n node-feature-discovery

  # Step 7: Verify the node got Intel GPU and NPU labels
  kubectl get node $(hostname) --show-labels | tr ',' '\n' | grep intel

  # Step 8: Install the Intel GPU device plugin
  kubectl apply -n intel-device-plugins -k "https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/gpu_plugin/overlays/nfd_labeled_nodes?ref=${RELEASE_VERSION}"

  # Step 9: Install the Intel NPU device plugin
  kubectl apply -n intel-device-plugins -k "https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/npu_plugin/overlays/nfd_labeled_nodes?ref=${RELEASE_VERSION}"
  ```

  Verify the GPU and NPU resources are advertised on nodes:
  ```bash
  kubectl get nodes -o json | jq '.items[] | {name: .metadata.name, gpu: .status.allocatable["gpu.intel.com/i915"], npu: .status.allocatable["npu.intel.com/accel"]}'
  ```
  > **Note:** If your node uses Intel Xe discrete GPUs (Arc), set `gpu:` to `.status.allocatable["gpu.intel.com/xe"]`.

> - K8s installation on single or multi node must be done as prerequisite to continue the following deployment. Note: The Kubernetes cluster is set up with `kubeadm`, `kubectl` and `kubelet` packages on single and multi nodes with `v1.30.2`.

Reviewer: ```suggestion
```
Move content to L6.

>   Refer to tutorials online to setup Kubernetes cluster on the web with host OS as Ubuntu 22.04 and/or Ubuntu 24.04.

Reviewer: ```suggestion
```
Move to L7.

> - For Helm installation, refer to [Helm website](https://helm.sh/docs/intro/install/)

Reviewer: ```suggestion
```
Move to L8.
